### PR TITLE
feat: Preliminary support for Python 3.13

### PR DESCRIPTION
**Pull request by sentrivana**
No RC yet, just beta releases, but we can already start setting stuff up.

3.13 release schedule: https://peps.python.org/pep-0719/#schedule

### CODE REVIEW

**Review by szokeasaurusrex — COMMENTED** [review]
Do these `allow-prereleases: true` have any affect on non-3.13 tests? I would be hesitant to merge this if so

File: sentry_sdk/utils.py [hunk 1]
```diff
@@ -611,7 +610,7 @@ def serialize_frame(
         )
 
     if include_local_variables:
-        rv["vars"] = copy(frame.f_locals)
+        rv["vars"] = frame.f_locals.copy()
```
Comment (antonpirker) [inline sentry_sdk/utils.py:613]: why this change?

**Review by antonpirker — APPROVED** [review]
🔥

**Comment by sentrivana** [comment]
> Do these allow-prereleases: true have any affect on non-3.13 tests? I would be hesitant to merge this if so

Will double check this.

> Edit: Just noticed the "Do not merge" label, is that still accurate?

No, removed now, thanks for noticing!

**Comment by sentrivana** [comment]
@szokeasaurusrex `allow-prereleases` will only kick in if the specified Python version is not generally available (all older Python versions are, so no effect there), see [here](https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#allow-pre-releases).

File: sentry_sdk/utils.py (hunk 1 above)
Comment (sentrivana) [reply sentry_sdk/utils.py:613]: [This commit](https://github.com/getsentry/sentry-python/pull/3200/commits/49b4cf328e115d5639e8bdc7b0978a8abf7c2fb6) has the details.

Comment (szokeasaurusrex) [reply sentry_sdk/utils.py:613]: https://github.com/getsentry/sentry-python/commit/49b4cf328e115d5639e8bdc7b0978a8abf7c2fb6

Comment (szokeasaurusrex) [reply sentry_sdk/utils.py:613]: ^ Basically, `frame.f_locals` is no longer a `dict` in 3.13

**Review by szokeasaurusrex — APPROVED** [review]
nice!
